lib/gias_api_client.rb

```diff
@@ -0,0 +1,56 @@
+# frozen_string_literal: true
+
+require "zip"
+
+class GiasApiClient
+  def get_files
+    data = fetch_extract
+    temp_files_from_zipped_response(data)
+  end
+
+private
+
+  def fetch_extract
+    client = Savon.client(wsdl: Rails.application.config.gias_api_schema)
+    res = client.call(
+      :get_extract,
+      message: { "tns:Id" => Rails.application.config.gias_extract_id },
+      wsse_auth: [Rails.application.config.gias_api_user, Rails.application.config.gias_api_password],
+      wsse_timestamp: true,
+    )
+    parts = parse_body(res.http)
```

Comment:
is our request using `http`? or `https`?

Not that I know what the hell `res.http` does, just wondering.

Reply:
It's not that. The initial request will be https to https://ea-edubase-api-prod.azurewebsites.net/edubase/schema/service.wsdl
, and subsequent requests dictated by the contents of that schema. This just return the HTTPI object that has header and body information